File: objective-c/DeviceOrientation/DeviceOrientation/ViewControllers/MainViewController.m
//
//  MainViewController.m
//  DeviceOrientation
//
//  Created by Alexandr Kirilov on 09.04.2021.
//  Copyright © 2021 Alexandr Kirilov. All rights reserved.
//

#import "MainViewController.h"

@interface MainViewController ()

@end

@implementation MainViewController

-(void) loadView {
    
    [super loadView];
    [self.view setBackgroundColor:__COLOR_GREEN];
}

-(void) viewDidLoad {
    
    [super viewDidLoad];
    
    _pPbutton = [UIButton buttonWithType:UIButtonTypeCustom];
    [_pPbutton addTarget:self action:@selector(mActionSetDeviceOrientation) forControlEvents:UIControlEventTouchUpInside];
    [_pPbutton setTitle:@"Change Orientation" forState:UIControlStateNormal];
    [_pPbutton setBackgroundColor:__COLOR_PURPLE];
    [_pPbutton setTranslatesAutoresizingMaskIntoConstraints:NO];
    [self.view addSubview:_pPbutton];
    [[_pPbutton.centerYAnchor constraintEqualToAnchor:self.view.centerYAnchor] setActive:YES];
    [[_pPbutton.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor] setActive:YES];
    [[_pPbutton.widthAnchor constraintEqualToConstant:self.view.bounds.size.width * 0.8] setActive:YES];
    [[_pPbutton.heightAnchor constraintEqualToConstant:self.view.bounds.size.height * 0.15] setActive:YES];
}

-(void) mActionSetDeviceOrientation {
    
    NSLog(@"Button clicked");
    [[UIDevice currentDevice] setValue: [NSNumber numberWithInteger: UIInterfaceOrientationLandscapeLeft] forKey:@"orientation"];
}

@end

File: swift/UIKit/UIKit UICollectionView/v2/UICollectionView/ViewControllers/MainViewController.swift
//
//  MainViewController.swift
//  UICollectionView
//
//  Created by Alexandr Kirilov on 18/03/2020.
//  Copyright © 2020 Alexandr Kirilov. All rights reserved.
//

import UIKit

class MainViewController: UIViewController {
	
	private var pCollectionView: UICollectionView = DemoCollectionView() as UICollectionView;

	override func viewDidLoad() {
		
		super.viewDidLoad();
		view.backgroundColor = __COLOR_BLUE;
		
		view.addSubview(pCollectionView);
		pCollectionView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor).isActive = true;
		pCollectionView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor).isActive = true;
		pCollectionView.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true;
		pCollectionView.centerYAnchor.constraint(equalTo: self.view.centerYAnchor).isActive = true;
		pCollectionView.heightAnchor.constraint(equalToConstant: 300.0).isActive = true;
	}
}
